### DISCUSSION TITLE
AOP原理 - StuartHub

### DISCUSSION THREAD

**Discussion by giscus** [StuartYang/StuartHub#3] [Announcements]
# AOP原理 - StuartHub

序言排队，洗头，理发，洗头，吹头 ，收费 除了理发外，其他的都是重复公共的流程，做成 公共代码，Spring 利用AOP将公共流程代码切入到主流代码中，每次主流代码的调用，都隐形的完成公共流程代码的执行。 AOP如何实现的利用代理，就和理发一样，用户理发时候通过助理完成一系列的服务操作 。代理可以伪装成“主体”，“骗”过调用方，在执行主体前夹带公共流程服务 。 这个 模型在AOP中：  助理：代理

https://stuarthub.us.kg/2023/12/08/AOP%E5%8E%9F%E7%90%86/

**Comment by StuartYang** [comment]
评论测评
